modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItRaftCommandLeftInLogUntilRestartTest.java

```diff
@@ -294,6 +303,44 @@ private void checkData(IgniteImpl ignite, Object[][] dataSet) {
                 new RuntimeException(IgniteStringFormatter.format("Cannot check a row {}", row), e);
             }
         }
+
+        transferLeadershipToLocalNode(ignite);
```

Comment:
Why do you need to transfer the leadership? I mean that transactions use primary replicas and not the leaders.

Reply:
Because currently we collaborate primarily with a leader, in the future this place should be reviewed. That is not the only place where we move a leader, but think about the primary.

Reply:
Why do you need to transfer the leadership? Could you please share test scenario?

Reply:
The transfer leader is needed to ensure that we are reading values from a specific node and that this works correctly (resolving write intents).

Reply:
Discussed in person. Let'it be transferLeader. WIll be covered in separate ticket with PD.transferPrimaryReplica